tfjs-backend-webgpu/src/kernels/Unary.ts

```diff
@@ -0,0 +1,249 @@
+    return log(a);`,
+    'implCPU': logImplCPU
+  },
+  [LogicalNot]: {
+    'shader': 'return f32(!(a >= 1.0));'
+  },
+  [Neg]: {
+    'shader': 'return -a;'
+  },
+  [Relu]: {
+    'shader': 'return max(a, 0.0);'
+  },
+  [Relu6]: {
+    'shader': 'return clamp(a, 0.0, 6.0);'
+  },
+  [Rsqrt]: {
+    'shader': 'return 1.0/sqrt(a);',
+    'implCPU': rsqrtImplCPU
+  },
+  [Sigmoid]: {
+    'shader': 'return 1.0 / (1.0 + exp(-1.0 * a));'
+  },
+  [Sin]: {
+    'shader': 'return sin(a);'
+  },
+  [Sinh]: {
+    'shader': `let e2x = exp(a);
+  return (e2x - 1.0 / e2x) / 2.0;`
+  },
+  [Sqrt]: {
+    'shader': 'return sqrt(a);'
+  },
+  [Square]: {
+    'shader': 'return a * a;'
+  },
+  [Tanh]: {
+    'shader': `let e2x = exp(-2.0 * abs(a));
+  return sign(a) * (1.0 - e2x) / (1.0 + e2x);`
+  },
+};
+
+const shaders: { [key: string]: string } = {};
+for (const kernelName in kernelInfo) {
+  shaders[kernelName] = kernelInfo[kernelName]['shader'];
+}
+shaders['Elu_vec4'] = `var resFloat = exp(a) - vec4<f32>(1.0);
+  if (a.r >= 0.0) {
+    resFloat.r = a.r;
+  }
+  if (a.g >= 0.0) {
+    resFloat.g = a.g;
+  }
+  if (a.b >= 0.0) {
+    resFloat.b = a.b;
+  }
+  if (a.a >= 0.0) {
+    resFloat.a = a.a;
+  }
+  return resFloat;`;
+shaders['Int'] = 'return f32(i32((a)));';
+shaders['Linear'] = 'return a;';
+shaders['Relu_vec4'] = `var resFloat = a * vec4<f32>(a >= vec4<f32>(0.0));
+  let isNaN = isNan(a);
+
+  if (isNaN.r) {
+    resFloat.r = a.r;
+  }
+  if (isNaN.g) {
+    resFloat.g = a.g;
+  }
+  if (isNaN.b) {
+    resFloat.b = a.b;
+  }
+  if (isNaN.a) {
+    resFloat.a = a.a;
+  }
+  return resFloat;`;
+shaders['Relu6_vec4'] = 'return clamp(a, vec4<f32>(0.0, 0.0, 0.0, 0.0), vec4<f32>(6.0, 6.0, 6.0, 6.0));';
+
+class UnaryProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: { x: number[] };
+  dispatch: [number, number, number];
+  variableNames = ['A'];
+  workGroupSize: [number, number, number];
+  kernelName: string;
+  size = true;
+
+  constructor(outputShape: number[], kernelName: string) {
+    // TODO(jiajia.qin@intel.com): Heuristically select a good work group size.
+    const workGroupSizeX = 128;
+    this.workGroupSize = [workGroupSizeX, 1, 1];
+    this.outputShape = outputShape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+      this.dispatchLayout, this.outputShape, this.workGroupSize);
+    this.kernelName = kernelName;
+    this.shaderKey = `unary_${kernelName}`;
+  }
+
+  getUserCode(): string {
+    return `
+      fn unaryKernel(a : f32) -> f32 {
+        ${shaders[this.kernelName]}
+      }
+      ${getMainHeaderAndGlobalIndexString()}
+        if (index < uniforms.size) {
+          let a = getAAtOutCoordsByGlobalIndex(index);
+          setOutputFlat(index, unaryKernel(a));
+        }
+      }
+      `;
+  }
+}
+
+type UnaryConfig = {
+  kernelName: string,
+  implCPU?: ImplCPU,
```

Comment:
If declare `implCPU?: ImplCPU` to `implCPU? ImplCPU | negImplCPU` and do similar in `kernelInfo`, does it work to merge `neg` to `unaryFunc`?